mcts.py

```diff
@@ -207,7 +209,24 @@ def incorporate_results(self, move_probabilities, value, up_to):
             self.revert_visits(up_to=up_to)
             return
         self.is_expanded = True
-        self.original_prior = self.child_prior = move_probabilities
+
+        '''
+        temp = move_probabilities * (1 - self.illegal_moves)
+        original = sum(move_probabilities)
+        new_s = sum(temp)
+
+        print ("{:.4f}   {:.4f}  {:.4f}".format(original - new_s, original, new_s))
+        assert abs(1 - original) < 1e-5
+        '''
+
+        # Zero out illegal moves.
+        move_probs = move_probabilities * (1 - self.illegal_moves)
+        scale = sum(move_probs)
+        if scale > 0:
```

Comment:
why would scale be <= 0?  No legal moves?

Reply:
Yes or a broken network file